README.md

```diff
@@ -1,126 +1,241 @@
+```bash
+cd ~/Downloads
+```
+Clone Leap using HTTPS...
+```bash
+git clone --recursive https://github.com/AntelopeIO/leap.git
+```
+...or SSH.
+```bash
+git clone --recursive git@github.com:AntelopeIO/leap.git
+```
+An HTTPS and SSH clone both result in the same thing, a folder named `leap` containg our source code, it doesn't matter which one you use. Navigate into that folder.
+```bash
+cd leap
+```
+Choose which [release tag](https://github.com/AntelopeIO/leap/releases) or [branch](#branches) you would like to build and check it out. If you don't know, use the [latest release](https://github.com/AntelopeIO/leap/releases/latest). For example, if you want to build release 3.1.2, you would check it out using its tag.
+```bash
+git checkout v3.1.2
+```
 
-#### Building Pinned Build Binary Packages
-In the directory `<leap src>/scripts` you will find the two scripts `install_deps.sh` and `pinned_build.sh`. If you haven't installed build dependencies then run `install_deps.sh`. Then run `pinned_build.sh <dependencies directory> <leap build directory> <number of jobs>`.
+Once you are on the branch or tag (release) you want to build, make sure everything is up-to-date.
+```bash
+git pull
+git submodule update --init --recursive
+```
 
-The dependencies directory is where the script will pull the C++ dependencies that need to be built with the pinned compiler for building the pinned binaries for binary packaging.
+#### Jobs Flag
+⚠️ **A Warning On Parallel Compilation Jobs (`-j` flag)** ⚠️
+When building C/C++ software, often the build is performed in parallel via a command such as `make -j "$(nproc)"` which uses all available CPU threads. However, be aware that some compilation units (`*.cpp` files) in Leap will consume nearly 4GB of memory to compile. Failures due to memory exhaustion will typically, but not always, manifest as compiler crashes. Using all available CPU threads may also prevent you from doing other things on your computer during compilation. Consider adjusting parallelization for these reasons.
 
-The binary package will be produced in the Leap build directory that was supplied.
+### Build
+We have two types of builds for Leap, "pinned" and "unpinned." The only difference is that pinned builds use specific versions for some dependencies hand-picked by Leap engineers, whereas unpinned builds use the default dependency versions available on the build system at the time. We recommend performing a pinned build to ensure the compiler and boost version remain the same between builds of different Leap versions. Leap requires these versions to remain the same, otherwise its state needs to be recovered from a portable snapshot or the chain needs to be replayed.
 
-#### Manual (non "pinned") Build Instructions
+For the curious, the "pinned" terminology comes from our days developing on macOS where you would run `brew pin` to lock one of the Leap dependencies to a specific version despite available upgrades.
```

Comment:
eh I'd recommend removing. Personally I don't recall homebrew pinning being the etymology of the "pinned build" nomenclature.

Reply:
Will remove. Maybe I made this up? Haha.

Reply:
Removed.